sites/shared/components/workbench/layout/print/plugin.js

```diff
@@ -10,68 +10,225 @@ const sizes = {
+ * sheetHeight: the height of the helper part
+ * boundary: should the helper part calculate its boundary?
+ * responsiveColumns: should the part make more columns if the pattern exceed its width? (for pages you want this, for fabric you don't)
+ * printStyle: hould the pages be rendered for printing or for screen viewing?
+ * */
+const basePlugin = ({
+  sheetWidth,
+  sheetHeight,
+  boundary=false,
+  partName="pages",
+  responsiveColumns=true,
+  printStyle=false
+}) => ({
   name,
   version,
   hooks: {
     postLayout: function(pattern) {
       // Add part
-      pattern.parts.pages = pattern.Part('pages')
+      pattern.parts[partName] = pattern.Part(partName)
       // Keep part out of layout
-      pattern.parts.pages.layout = false
+      pattern.parts[partName].layout = false
       // But add the part to the autoLayout property
-      pattern.autoLayout.parts.pages = {
+      pattern.autoLayout.parts[partName] = {
         move: { x: 0, y: 0 }
       }
       // Add pages
-      const { macro } = pattern.parts.pages.shorthand()
-      const { height, width } = pattern
-      macro('addPages', { size, orientation, height, width })
+      const { macro } = pattern.parts[partName].shorthand()
+      let { height, width } = pattern
+      if (!responsiveColumns) width = sheetWidth;
+      if (pattern.settings.layout?.topLeft) {
+        height += pattern.settings.layout.topLeft.y
+        responsiveColumns && (width += pattern.settings.layout.topLeft.x)
+      }
+
+      macro('addPages', { size: [sheetWidth, sheetHeight], height, width })
+
+      if (boundary) pattern.parts[partName].boundary();
     }
   },
   macros: {
+    /** draft the pages */
     addPages: function(so) {
-      const ls = so.orientation === 'landscape'
-      const w = sizes[so.size][ls ? 1 : 0]
-      const h = sizes[so.size][ls ? 0 : 1]
+      const [h,w] = so.size
       const cols = Math.ceil(so.width / w)
       const rows = Math.ceil(so.height / h)
-      const { points, Point, paths, Path } = this.shorthand()
+      const { points, Point, paths, Path, macro } = this.shorthand()
       let x = 0
       let y = 0
       let count = 0
       for (let row=0;row<rows;row++) {
         x=0
         for (let col=0;col<cols;col++) {
           count++
-          points[`_pages__row${row}-col${col}-tl`] = new Point(x,y)
-          points[`_pages__row${row}-col${col}-tr`] = new Point(x+w,y)
-          points[`_pages__row${row}-col${col}-br`] = new Point(x+w,y+h)
-          points[`_pages__row${row}-col${col}-bl`] = new Point(x,y+h)
-          points[`_pages__row${row}-col${col}-circle`] = new Point(x+w/2,y+h/2-24)
+          const pageName = `_pages__row${row}-col${col}`
+          points[`${pageName}-tl`] = new Point(x,y)
+          points[`${pageName}-tr`] = new Point(x+w,y)
+          points[`${pageName}-br`] = new Point(x+w,y+h)
+          points[`${pageName}-bl`] = new Point(x,y+h)
+          points[`${pageName}-circle`] = new Point(x+w/2,y+h/2)
```

Comment:
Should this be suppressed when `printStyle` is falsy?
I'm not the biggest fan of those big circled letters staying on the printout, considering we add the info on the sides in a more discrete way.

Reply:
Let's leave it for now. We can revisit this later.